# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1452937967 -39600
#      Sat Jan 16 20:52:47 2016 +1100
# Node ID a1aa52bff0797d43c6c358045f1903a2472b5be8
# Parent  dc325209923221a76a4b8dc724d6c3f935eb3730
Allow bsxfun to work for sparse and type conversion (bug #45219, bug #40089).

* bsxfun.cc (Fbsxfun): Initialise the fall-back variable C for sparse too.

diff --git a/libinterp/corefcn/bsxfun.cc b/libinterp/corefcn/bsxfun.cc
--- a/libinterp/corefcn/bsxfun.cc
+++ b/libinterp/corefcn/bsxfun.cc
@@ -513,16 +513,21 @@ dimensionality as the other array.\n\
                       else if BSXINIT(uint32NDArray, "uint32", uint32)
                       else if BSXINIT(uint64NDArray, "uint64", uint64)
                       else
                         {
                           C = tmp(0);
                           C = C.resize (dvc);
                         }
                     }
+                  else  // Skip semi-fast path for sparse matrices
+                    {
+                      C = tmp (0);
+                      C = C.resize (dvc);
+                    }
                 }
               else
                 {
                   update_index (ra_idx, dvc, i);
 
                   if (have_FloatNDArray
                       || have_FloatComplexNDArray)
                     {
